docs/containers-storage-architecture.md

```diff
@@ -0,0 +1,76 @@
+## containers-storage 1 "May 2024"
+
+## NAME
+containers-storage-architecture - The implementation of containers-storage 
+
+NOTE: These are implementation details subject to change. This document
+is intended to be used only as an aid for debugging and development
+of the containers-storage system. Do *not* directly access the filesystem
+components below from external code, only from the Go API.
+
+## DESCRIPTION
+The *containers/storage* library manages the following object types:
+
+- layer: A filesystem tree, along with metadata
+- image: A reference to a layer, along with metadata
+- container: A layer whose parent is an image
+
+## Layer storage
```

Comment:
Discusssion of graph drivers?

Reply:
Would make sense to add in theory but I personally don't care much since I think we're in a place where there is only the `overlay` driver that matters, right?